File: Rust/src/solution/distance_between_bus_stops.rs
use crate::solution::Solution;

impl Solution {
    pub fn distance_between_bus_stops(distance: Vec<i32>, start: i32, destination: i32) -> i32 {
        let (start, destination) = (start.min(destination), start.max(destination));
        let (s, d) = (start as usize, destination as usize);
        let clockwise: i32 = distance[s..d].iter().sum();
        let total: i32 = distance.iter().sum();
        clockwise.min(total - clockwise)
    }
}

File: Dart/lib/distanceBetweenBusStops.dart
class Solution {
  int distanceBetweenBusStops(List<int> distance, int start, int destination) {
    final total = distance.reduce((a, b) => a + b);
    final (min, max) = start < destination ? (start, destination) : (destination, start);
    final sum = distance.sublist(min, max).reduce((a, b) => a + b);
    return (sum < (total - sum)) ? sum : (total - sum);
  }
}
